lib/fluent/plugin_id.rb

```diff
@@ -59,5 +64,17 @@ def plugin_id
         "object:#{object_id.to_s(16)}"
       end
     end
+
+    def plugin_root_dir
+      return @_plugin_root_dir if @_plugin_root_dir
+      return nil unless system_config.root_dir
```

Comment:
No, it's not about users.
This code is called by plugins, and plugin should check whether `plugin_root_dir` is available or not by return value, and raise errors with messages which shows how to configure that plugin.